### PR TITLE
Refactor AppSource Gallery tests to use test library

### PR DESCRIPTION
**Pull request by blrobl**
#### Work Item(s) 
Fixes [AB#501895](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/501895)

### CODE REVIEW

File: src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.TestLibraries.Apps.AppSource;
+
+using System.Environment.Configuration;
+using System.RestClient;
+using System.Apps.AppSource;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
```
Comment (grobyns) [inline src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al]:
```suggestion
/// Library for managing AppSource product retrieval and usage.
```

File: src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+/// </summary>
+codeunit 132910 "AppSource Product Manager Test" implements "AppSource Product Manager Dependencies"
```
Comment (grobyns) [inline src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al]:
```suggestion
codeunit 132910 "AppSrc Product Mgr. Test Impl." implements "AppSource Product Manager Dependencies"
```

Comment (grobyns) [inline src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al]: This code would be much easier to understand if you split it into 2 codeunits: 
- one is the implementation of the interface, which is really only used for mocking some env. stuff in the tests
- the other is the 'test library' which would be helper methods to ease testing and some wrappers around internal functionality

Comment (blrobl) [reply src/System Application/Test Library/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al]: Agree, did the same for the module itself

**Review by SBalslev — APPROVED** [review]

File: src/System Application/Test Library/AppSource Gallery/src/AppSourceMockDepsProvider.Codeunit.al [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.TestLibraries.Apps.AppSource;
+
+using System.Environment.Configuration;
+using System.RestClient;
+using System.Apps.AppSource;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
```
Comment (grobyns) [inline src/System Application/Test Library/AppSource Gallery/src/AppSourceMockDepsProvider.Codeunit.al]:
```suggestion
/// Library for managing AppSource product retrieval and usage.
```

Comment (grobyns) [reply src/System Application/Test Library/AppSource Gallery/src/AppSourceMockDepsProvider.Codeunit.al]: not sure this text is accurate, same as other object... one of them might need an update

File: src/System Application/Test Library/AppSource Gallery/src/AppSrcProductMgrTestImpl.Codeunit.al [hunk 4]
```diff
@@ -0,0 +1,94 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.TestLibraries.Apps.AppSource;
+
+using System.Apps.AppSource;
+
+/// <summary>
+/// Library for managing AppSource product retrival and usage.
```
Comment (grobyns) [inline src/System Application/Test Library/AppSource Gallery/src/AppSrcProductMgrTestImpl.Codeunit.al]:
```suggestion
/// Library for managing AppSource product retrieval and usage.
```

**Review by grobyns — DISMISSED** [review]

**Review by SBalslev — APPROVED** [review]

**Review by SBalslev — APPROVED** [review]

**Review by grobyns — APPROVED** [review]

**Pull request merged**